aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -0,0 +1,55 @@
+/// This module defines structs and methods to initialize VM configurations,
+/// including different costs of running the VM.
+module aptos_framework::gas_schedule {
+    use std::error;
+    use std::string::String;
+
+    use aptos_framework::reconfiguration;
+    use aptos_framework::system_addresses;
+    use aptos_framework::timestamp;
+    use aptos_framework::util::from_bytes;
+
+    /// Error with config
+    const ECONFIG: u64 = 0;
+    /// The provided gas constants were inconsistent.
+    const EGAS_CONSTANT_INCONSISTENCY: u64 = 1;
+
+    struct GasEntry has store, copy, drop {
+        key: String,
+        val: u64,
+    }
+
+    struct GasSchedule has key, copy, drop {
+        entries: vector<GasEntry>
+    }
+
+    public fun initialize(account: &signer, gas_schedule_blob: vector<u8>) {
+        timestamp::assert_genesis();
+        system_addresses::assert_aptos_framework(account);
+
+        assert!(
+            !exists<GasSchedule>(@aptos_framework),
+            error::already_exists(ECONFIG)
+        );
+
+        // TODO(Gas): check if gas schedule is consistent
+
+        let gas_schedule: GasSchedule = from_bytes(gas_schedule_blob);
+
+        move_to(account, gas_schedule);
+    }
+
+    public entry fun set_gas_schedule(account: &signer, gas_schedule_blob: vector<u8>) acquires GasSchedule {
+        timestamp::assert_operating();
+        system_addresses::assert_core_resource(account);
+
+        assert!(exists<GasSchedule>(@aptos_framework), error::not_found(ECONFIG));
+
+        // TODO(Gas): check if gas schedule is consistent
+
+        let gas_schedule = borrow_global_mut<GasSchedule>(@aptos_framework);
+        *gas_schedule = from_bytes(gas_schedule_blob);
+
+        reconfiguration::reconfigure();
```

Comment:
Why is this necessary? Does the adapter also rely on the new epoch event to reload gas schedule? If so, it seems like overloading of that event. Can we use a separate, new event?

Reply:
It used to be that any on-chain config change would require a reconfiguration. Not totally sure if this is still the case. @zekun000 any ideas?

Reply:
I think this is governance responsibilities to call.

Reply:
This can be done as a follow up but the part that's missing for me is how the Rust side syncs when the gas schedule changes on chain. If it's through reconfiguration, I think we should separate it out. Triggering reconfiguration can impact validator set and rewards calculation. We should avoid doing that for every single change if possible.

Reply:
it's the correct use of the new epoch event/reconfiguration because it needs to trigger a new epoch (for validators to align on the same gas schedule)